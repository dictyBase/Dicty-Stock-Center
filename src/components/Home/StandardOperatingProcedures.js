// @flow
import React from "react"
import { withStyles } from "@material-ui/core/styles"
import styles from "./homeStyles"

type Props = {
  /** Material-UI styling */
  classes: {
    panelGray: string,
    sopLink: string,
  },
}

/**
 * StandardOperatingProcedures displays the SOP link (Box PDF).
 */

const StandardOperatingProcedures = (props: Props) => {
  const { classes } = props

  return (
    <div className={classes.panelGray}>
      <a
        className={classes.sopLink}
        href="https://northwestern.box.com/s/p0f8m70whgiuib2u0wt8gtn497ncmq8i"
        rel="noopener noreferrer"
        target="_blank">
        Standard Operating Procedures
      </a>
    </div>
  )
}

export default withStyles(styles)(StandardOperatingProcedures)
