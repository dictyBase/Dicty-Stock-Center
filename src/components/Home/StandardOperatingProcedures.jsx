import React, { Component } from 'react'
import LinkList from '../LinkList'
import { PanelGray } from 'styles'

export default class Materials extends Component {
    displayName = 'links to stock catalogs'
    render() {
        const procedures = [
            {name: 'Standard Operating Procedures', to: '#', routerAware: true}
        ]
        return (
            <PanelGray>
                <LinkList list={ procedures }/>
            </PanelGray>
        )
    }
}

