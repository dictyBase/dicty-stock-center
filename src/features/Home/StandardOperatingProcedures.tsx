import React from "react"
import Box from "@material-ui/core/Box"
import useStyles from "./homeStyles"

/**
 * StandardOperatingProcedures displays the SOP link (Box PDF).
 */

const StandardOperatingProcedures = () => {
  const classes = useStyles({ panelBackground: "grey" })

  return (
    <Box className={classes.panel}>
      <a
        className={classes.sopLink}
        href="https://northwestern.box.com/s/p0f8m70whgiuib2u0wt8gtn497ncmq8i"
        rel="noopener noreferrer"
        target="_blank">
        Standard Operating Procedures
      </a>
    </Box>
  )
}

export default StandardOperatingProcedures
