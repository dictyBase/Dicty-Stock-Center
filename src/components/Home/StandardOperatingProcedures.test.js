import React from 'react'
import ReactDOM from 'react-dom'
import renderer from 'react-test-renderer'
import StandardOperatingProcedures from './StandardOperatingProcedures'

it('renders without crashing', () => {
    const div = document.createElement('div')
    ReactDOM.render(<StandardOperatingProcedures />, div)
})

test('matching a snapshot of StandardOperatingProcedures', () => {
    const component = renderer.create(<StandardOperatingProcedures />)
    let tree = component.toJSON()
    expect(tree).toMatchSnapshot()
})
