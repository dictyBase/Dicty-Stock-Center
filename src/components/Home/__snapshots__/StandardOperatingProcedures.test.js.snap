// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matching a snapshot of StandardOperatingProcedures 1`] = `
<div
  className="sc-jzJRlG itJXei"
>
  <a
    href="https://northwestern.box.com/s/p0f8m70whgiuib2u0wt8gtn497ncmq8i"
    rel="noopener noreferrer"
    target="_blank"
  >
    Standard Operating Procedures
  </a>
  Â 
  <span
    aria-hidden={true}
    className="fa fa-file-pdf-o"
  />
</div>
`;
